component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcCallSink.java

```diff
@@ -0,0 +1,256 @@
+    public void publish(Object payload, DynamicOptions dynamicOptions, State state)
+            throws ConnectionUnavailableException {
+        if (isDefaultMode) {
+            if (!(payload instanceof String)) {
+                throw new SiddhiAppRuntimeException(siddhiAppContext.getName() + ": Payload should be of type String " +
+                        "for default EventService but found " + payload.getClass().getName());
+            }
+            Event sequenceCallRequest = Event.newBuilder().setPayload((String) payload).build();
+            EventServiceFutureStub currentFutureStub = futureStub;
+
+            if (sequenceName != null || headersOption != null) {
+                Metadata header = new Metadata();
+                String headers = "";
+                if (sequenceName != null) {
+                    headers += "'sequence:" + sequenceName + "'";
+                    if (headersOption != null) {
+                        headers += ",";
+                    }
+                }
+                if (headersOption != null) {
+                    headers +=  headersOption.getValue(dynamicOptions);
+                }
+                Metadata.Key<String> key =
+                        Metadata.Key.of(GrpcConstants.HEADERS, Metadata.ASCII_STRING_MARSHALLER);
+                header.put(key, headers);
+                currentFutureStub = MetadataUtils.attachHeaders(futureStub, header);
+            }
+            ListenableFuture<Event> futureResponse =
+                    currentFutureStub.process(sequenceCallRequest);
+            Futures.addCallback(futureResponse, new FutureCallback<Event>() {
+                @Override
+                public void onSuccess(Event result) {
+                    GrpcSourceRegistry.getInstance().getGrpcCallResponseSourceSource(sinkID).onResponse(result);
+                }
+
+                @Override
+                public void onFailure(Throwable t) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug(siddhiAppContext.getName() + ": " + t.getMessage());
+                    }
+                }
+            }, MoreExecutors.directExecutor());
+        } else {
+            //todo: handle publishing to generic service
+        }
+    }
+
+    /**
+     * This method will be called before the processing method.
+     * Intention to establish connection to publish event.
+     * @throws ConnectionUnavailableException if end point is unavailable the ConnectionUnavailableException thrown
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        this.channel = ManagedChannelBuilder.forTarget(address).usePlaintext().build();
+        this.futureStub = EventServiceGrpc.newFutureStub(channel);
+        if (!channel.isShutdown()) {
+            logger.info(streamID + " has successfully connected to " + url);
```

Comment:
streamID is not conected the` grpc... on streamID` is

Reply:
if channel.isShutdown() then what to do?? dont we need  an error ?

Reply:
I moved the channel building part into connect(). Previously did this in init() thats why I added the if check. Will remove the if check but keep the modified log